hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -858,15 +860,29 @@ private static int parsePartNumberMarker(String partNumberMarker) {
     return partMarker;
   }
 
-  private static long parseOzoneDate(String ozoneDateStr) throws OS3Exception {
+  // If the return is empty, the precondition should be ignored
```

Comment:
Can you please update this comment and others in the method.
```suggestion
  // Parses date string and return long representation. Returns an empty if DateStr is null or invalid. Date in the future are considered invalid.
```